Src/Base/AMReX_Arena.cpp

```diff
@@ -11,7 +11,17 @@
 #include <AMReX_ParmParse.H>
 #include <AMReX_Gpu.H>
 
+#ifdef _WIN32
```

Comment:
I think it's the right way to support the system compilers (XCode's Clang fork on macOS and MSVC on Windows).

Nevertheless, where specific to a stdlib & runtime system, I would replace `__APPLE__` and `_WIN32` macros with slightly more specific Apple+Clang guards and MSVC guards (`if defined(_MSC_VER) && !defined(__INTEL_COMPILER)`).

Background: there are GNU-based compilers for Windows, too such as Cygwin and MSYS2 that some scientific projects use which rely on e.g. a full POSIX.

On macOS, one might need to potentially differentiate between vanilla clang, xcode clang and gcc, too - but let's just add this when it's actually needed.

Reply:
```In gcc for Cygwin, _WIN32 is only defined when you use the -mwin32 gcc command line options. This is because Cygwin is supposed to be a POSIX emulation environment in the first place and defining _WIN32 confuses some programs which think that they have to make special concessions for a Windows environment which Cygwin handles automatically.```

This is from https://cygwin.com/faq/

Reply:
As for Mac, there are a few places with `__APPLE__` for two purposes.  One is backtrace support.  I don't have access to Mac.  So I used someone's Mac and added backtrace support.  I think it works with gcc and I don't know if it works with clang.  The other is some cuda stuff does not work on Mac.

Reply:
Sounds good then! CUDA is deprecated on macOS anyway and removed in CUDA11+.